fix: Patch remaining runtime and cleanup issues

p1carats opened the pull request:
### Description

This PR continues the work initiated in [#1808](https://github.com/jorenn92/Maintainerr/pull/1808), addressing several additional runtime and cleanup issues that were identified during testing.

Specifically:

- Prevents a crash when the Overseerr/Jellyseerr URL is left empty.
- Cleans up various event listeners on component unmount to avoid memory leaks.
- Cancels untracked Plex-related Axios requests to avoid unexpected behavior on component teardown.

These changes contribute to the broader goal of stabilizing long-running sessions and reducing memory bloat in edge scenarios.

### Related issue

Follow-up to #1808 — no formal issue created.

### Checklist

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) document.
- [X] I have performed a self-review of my code.
- [X] I have linted and formatted my code.
- [X] My changes generate no new warnings.
- [X] New and existing unit tests pass locally with my changes.

### How to test

Run the app normally and test the following:

- Leave the Overseerr or Jellyseerr URL blank (which is often the case on first boot or when no data is configured). One of the two is typically empty, as using both simultaneously is not a common use case. Ensure no runtime crash occurs in this scenario.
- Trigger navigation across tabs and validate that event listeners are correctly cleaned up (e.g., no memory bloat, no duplicate events).
- Open the Plex tab, wait for API calls, then switch views — verify no cancelled Axios warning or memory leaks are triggered.

### Additional context

This is part of an ongoing effort to tighten the app's memory and lifecycle management.
While these patches resolve some remaining edge cases missed in #1808, there may still be other latent issues. A deeper audit may be worthwhile, but this PR might at least help improve long-session reliability and reduce noisy errors or warnings.

As previously mentioned, the removed commit related to timer handling and cleanup is still pending reintroduction. It was causing unexpected behavior and needs further refinement before being safely merged. This follow-up PR focuses on other isolated fixes, and that timer-related patch will be revisited in a future contribution.

*A follow-up hotfix or patch release is again recommended after merging.*

benscobie commented on ui/src/components/Settings/Plex/index.tsx:
```diff
@@ -213,6 +213,8 @@ const PlexSettings = () => {
   const verifyToken = (token?: string) => {
     const authToken = token || settingsCtx.settings.plex_auth_token
     if (authToken) {
+      const source = axios.CancelToken.source()
```
This method of cancelling is deprecated, AbortController is the way to go now:

```suggestion
      const controller = new AbortController();
```

You also need to pass this into axios for it to work

benscobie commented on ui/src/components/Settings/Plex/index.tsx:
```diff
@@ -226,6 +228,11 @@ const PlexSettings = () => {
+
+      // Cancel the request if component unmounts
+      return () => {
+        source.cancel('Component unmounted')
```
```suggestion
        controller.abort()
```

benscobie commented on ui/src/components/Settings/Logs/index.tsx:
```diff
@@ -136,21 +136,27 @@ const Logs = () => {
     }
 
     return () => {
+      es.removeEventListener('log', handleLog)
```
Oops, good spot

benscobie commented:
I'm interested in your workflow for tracking memory leaks. My first thought would be to use the memory tool in dev tools, but comparing snapshots doesn't seem to be the easiest?

p1carats commented on ui/src/contexts/events-context.tsx:
```diff
@@ -44,7 +44,11 @@ export const useEvent = <T,>(
-      listenerAdded.current(JSON.parse(ev.data))
+      try {
+        listenerRef.current(JSON.parse(ev.data))
+      } catch (error) {
```
I actually renamed `listenerAdded` to `listenerRef` to make the naming a bit more consistent and reflective of its purpose. And yeah, I wrapped it in a try/catch just to be safe — I didn’t run into a specific error as far as I remember, but I stumbled upon it by chance and figured it wouldn’t hurt to guard against any edge cases. Better safe than sorry!

p1carats commented:
Just pushed an update! I applied your suggestions regarding the deprecated pattern in the Plex Axios request — thanks again for catching that. I’ve added you as a co-author for that commit @benscobie 🙂

Also added a new commit with a few small fixes, mostly React minor warnings related to controlled vs uncontrolled form states.

Let me know if anything else needs tweaking!

benscobie commented:
> Full transparency: I’m running Maintainerr outside of Docker right now, in a pretty tightly constrained environment with a hard ulimit on memory. The app would mostly work, but I kept getting random crashes tied to some glib issue from a Node module. That’s what got me digging in the first place (and explains my contributions to the release-workflow branch too, haha).

> I’ve just been putting the app under memory pressure and watching how it behaves. 

Are you running the server + your client (browser) on the same memory-constrained system? The reason I ask as if you're only talking about the server-side, then I am surprised you saw any reduction in memory with the changes made as they're all targeting the client.

benscobie commented on ui/src/components/Common/MediaCard/index.tsx:
```diff
@@ -132,6 +132,7 @@ const MediaCard: React.FC<IMediaCard> = ({
               alt=""
               src={`https://image.tmdb.org/t/p/w300_and_h450_face${image}`}
               fill
+              sizes="9.375rem"
```
How did you come up with the number? I'm looking at the generated srcset and I can't see any relation to this 😆 

I also noticed a problem with the CachedImage implementation. It is referencing settingsCtx which only gets populated on the settings pages by the looks of it... So this leads to unoptimized being true as it's defaulted to an empty object. Of course if you swap to a settings page and then back, it'll then reference the cached image instead.

I think we could move that fetch and set of settings to the layout so we can get access to the context everywhere. The other alternative is loading the settings within the CachedImage component, and use a fetch library that de-dupes requests like react-query (already a dependency). The settings provider should be moved to the root settings page (\ui\src\components\Settings\index.tsx) in this case though, as it's only used in that tree.

benscobie replied:
I swear they've changed the sharp stuff several times now. I'm sure at some point they started shipping it as part of their package. It's the first time I've seen squoosh mentioned as well.

I've used sharp in the past but we ended up disabling image optimization entirely as it was hurting performance at times. You can certainly give it a go with your setup and see if it's any better.

We can leave the cached image fix I mentioned for a different PR 😊

benscobie replied:
Oh and on the sizes change, I'll need to check if that's a large enough value, as that CSS was using minmax. 1fr could be larger just before the next breakpoint hits when the cards are in a single column.

benscobie replied:
I would probably go with the default of `100vw` here for now. As the component is referenced in a few places that contain them in a grid, with varying different breakpoints, I think it will be a bit of a pain to generate a sensible list. The smallest I saw a card go was around 1/2 the size of the source image (300x450). With `sizes=100vw`, I'm getting 24KB~ optimised files, with the src image being 37KB~

p1carats commented:
I've updated the size value to `100vw` as requested and force-pushed the changes. The PR looks good on my end!

benscobie reviewed (approved):
Thank you 🚀

jorenn92 commented:
:tada: This PR is included in version 2.18.1 :tada:

The release is available on [GitHub release](https://github.com/jorenn92/Maintainerr/releases/tag/v2.18.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
